test/sslapitest.c

```diff
@@ -9831,6 +9876,255 @@ static int test_load_dhfile(void)
 #endif
 }
 
+#if !defined(OSSL_NO_USEABLE_TLS1_3) && !defined(OPENSSL_NO_COMP) && (defined(ZLIB) || !defined(OPENSSL_NO_BROTLI) || !defined(OPENSSL_NO_ZSTD))
+/*
+ * Test 0 = Skip compression
+ * Test 1 = libssl does the compression
+ * Test 2 = error out during compression
+ * Test 3 = libssl does compression, client authentication
+ * Test 4 = app does the compression
+ */
+/* This callback just says to skip compression */
+static int cert_comp_cb0(SSL *ssl, uint16_t alg, void* arg)
+{
+    *(int*)arg = alg;
+    return 0;
+}
+/* This callback just says to do the compression */
+static int cert_comp_cb1(SSL *ssl, uint16_t alg, void* arg)
+{
+    *(int*)arg = alg;
+    return alg;
+}
+/* This callback just says to error out */
+static int cert_comp_cb2(SSL *ssl, uint16_t alg, void* arg)
+{
+    return -1;
+}
+static int cert_comp_cb4(SSL *ssl, uint16_t alg, void* arg)
+{
+    unsigned char *cert_data = NULL;
+    size_t cert_length = 0;
+    unsigned char *comp_data = NULL;
+    size_t comp_length = 0;
+    int retval = 0;
+    COMP_METHOD *method;
+    COMP_CTX *comp_ctx = NULL;
+    int comp_result;
+
+    if (!TEST_ptr(cert_data = SSL_get_cert_to_compress(ssl, &cert_length)))
+        goto err;
+
+    /* Assume 50% - just for testing */
+    comp_length = cert_length + (cert_length >> 1);
+    if (!TEST_ptr(comp_data = OPENSSL_zalloc(comp_length)))
+        goto err;
+
+    switch (alg) {
+#ifndef OPENSSL_NO_BROTLI
+    case TLSEXT_comp_cert_brotli:
+        method = COMP_brotli_oneshot();
+        break;
+#endif
+#if defined(ZLIB)
+    case TLSEXT_comp_cert_zlib:
+        method = COMP_zlib();
+        break;
+#endif
+#ifndef OPENSSL_NO_ZSTD
+    case TLSEXT_comp_cert_zstd:
+        method = COMP_zstd_oneshot();
+        break;
+#endif
+    default:
+        goto err;
+    }
+
+    if (!TEST_ptr(method) || !TEST_ptr(comp_ctx = COMP_CTX_new(method)))
+        goto err;
+
+    comp_result = COMP_compress_block(comp_ctx,
+                                      comp_data, comp_length,
+                                      cert_data, cert_length);
+    if (!TEST_int_gt(comp_result, 0))
+        goto err;
+
+    comp_length = comp_result;
+
+    /* Did we expand? This cert shouldn't expand */
+    if (TEST_size_t_ge(comp_length, cert_length))
```

Comment:
This should use `!TEST_size_t_lt()` instead. Otherwise the test logging gets screwed up. `TEST_*` macros are expected to pass normally. If they fail it prints extra logging information.